### PR TITLE
#151946827 Fixes and adds API tests

### PR DESCRIPTION
**Pull request by CeciliaCaroline**
#### What does this PR do?
It fixes and adds tests for the API app. New tests have been added to the existing tests and failing tests have been fixed.
 

#### How should this be manually tested?
After cloning the repo, cd into it and RUN `python manage.py test hc/api`

#### Any background context you want to provide?
 The app hc/api had its unit tests but some of these tests are incomplete and need fixing and some asserts added.

#### What are the relevant pivotal tracker stories?
#151946827

### CODE REVIEW

**Review by VincentHokie — APPROVED** [review]
We did some good stuff here @CeciliaCaroline , don't see anything worth changing

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,5 @@ hc/local_settings.py
 static-collected
 
 .DS_Store
-static/.DS_Store
\ No newline at end of file
+static/.DS_Store
```
Comment (ianoti) [inline .gitignore]: could generalise this by `*/.DS_Store`

File: hc/api/tests/test_badge.py [hunk 2]
```diff
@@ -13,12 +13,19 @@ def setUp(self):
...
+        
     def test_it_returns_svg(self):
         sig = base64_hmac(str(self.alice.username), "foo", settings.SECRET_KEY)
         sig = sig[:8].decode("utf-8")
         url = "/badge/%s/%s/foo.svg" % (self.alice.username, sig)
 
         r = self.client.get(url)
         ### Assert that the svg is returned
+
+        # self.assertEqual(r.status_code, 400)
```
Comment (ianoti) [inline hc/api/tests/test_badge.py]: remove commented out dead code

File: hc/api/tests/test_create_check.py [hunk 3]
```diff
@@ -16,7 +16,10 @@ def post(self, data, expected_error=None):
 
         if expected_error:
             self.assertEqual(r.status_code, 400)
+            
             ### Assert that the expected error is the response error
+            self.assertEqual(
+                json.loads(str(r.content.decode('utf-8') ))["error"], expected_error)
```
Comment (ianoti) [inline hc/api/tests/test_create_check.py]: small nitpick here of PEP8 styling remove space in parentheses

File: hc/api/tests/test_create_check.py [hunk 4]
```diff
@@ -46,36 +51,101 @@ def test_it_works(self):
...
-        self.assertEqual(r['status_code'], 201)
+        r  = r.json()
+        self.assertEqual(r['error'], "wrong api_key")
 
     def test_it_handles_missing_request_body(self):
         ### Make the post request with a missing body and get the response
-        r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
-        self.assertEqual(r['status_code'], 400)
+        r = self.post({})
+        # r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
```
Comment (ianoti) [inline hc/api/tests/test_create_check.py]: remove this commented out placeholder test

File: hc/api/tests/test_create_check.py [hunk 5]
```diff
@@ -46,36 +51,101 @@ def test_it_works(self):
...
+                             content_type="application/json")
+        self.assertEqual(r.status_code, 400)
+        r = r.json()
         self.assertEqual(r["error"], "could not parse request body")
 
     def test_it_rejects_wrong_api_key(self):
-        self.post({"api_key": "wrong"},
+        r = self.post({"api_key": "wrong"},
                   expected_error="wrong api_key")
+        r = r.json()
```
Comment (ianoti) [inline hc/api/tests/test_create_check.py]: the following assertions are unneccessary they are handled from within the post method defined

File: hc/api/tests/test_create_check.py [hunk 6]
```diff
@@ -46,36 +51,101 @@ def test_it_works(self):
...
+        self.assertEqual(r["error"], "wrong api_key")
+
 
     def test_it_rejects_non_number_timeout(self):
-        self.post({"api_key": "abc", "timeout": "oops"},
+        r = self.post({"api_key": "abc", "timeout": "oops"},
                   expected_error="timeout is not a number")
 
+        r = r.json()
```
Comment (ianoti) [inline hc/api/tests/test_create_check.py]: same as before this assertion is handled within the post method defined in this file

File: hc/api/tests/test_create_check.py [hunk 7]
```diff
@@ -46,36 +51,101 @@ def test_it_works(self):
...
+        self.assertEqual(r["error"], "timeout is not a number")
+
     def test_it_rejects_non_string_name(self):
-        self.post({"api_key": "abc", "name": False},
+        r = self.post({"api_key": "abc", "name": False},
                   expected_error="name is not a string")
+        
+        r = r.json()
+        self.assertEqual(r["error"], "name is not a string")
```
Comment (ianoti) [inline hc/api/tests/test_create_check.py]: same here assertion is unnecessary.

File: hc/api/tests/test_create_check.py [hunk 8]
```diff
@@ -46,36 +51,101 @@ def test_it_works(self):
...
+            # matched the keys in the channels we created up there
+            assert channel.code in channels
+
+    
     ### Test for the 'timeout is too small' and 'timeout is too large' errors
+    def test_it_rejects_very_small_timeouts(self):
+        r = self.post({"api_key": "abc", "name": "abc", "timeout":10},
+                  expected_error="timeout is too small")
+        
+        r = r.json()
```
Comment (ianoti) [inline hc/api/tests/test_create_check.py]: same as above should be handled from within post method

File: hc/api/tests/test_create_check.py [hunk 9]
```diff
@@ -46,36 +51,101 @@ def test_it_works(self):
...
+        self.assertEqual(r["error"], "timeout is too small")
+
+    
+    def test_it_rejects_very_large_timeouts(self):
+        r = self.post({"api_key": "abc", "name": "abc", "timeout":100000000},
+                  expected_error="timeout is too large")
+        
+        r = r.json()
```
Comment (ianoti) [inline hc/api/tests/test_create_check.py]: same as above

Comment (ianoti) [reply hc/api/tests/test_create_check.py]: the assertions are unnecessary.

File: hc/api/tests/test_list_checks.py [hunk 10]
```diff
@@ -34,14 +33,33 @@ def get(self):
...
         r = self.get()
         ### Assert the response status code
+        assert r.status_code == 200
 
         doc = r.json()
         self.assertTrue("checks" in doc)
 
         checks = {check["name"]: check for check in doc["checks"]}
         ### Assert the expected length of checks
+        assert len(checks) == 2
```
Comment (ianoti) [inline hc/api/tests/test_list_checks.py]: can you change all the `assert` statements to use `self.assert` convention

File: hc/api/tests/test_list_checks.py [hunk 11]
```diff
@@ -34,14 +33,33 @@ def get(self):
...
+        assert r['ping_url'] == "http://localhost:8000/ping/" + str(self.a1.code)
+        assert r['pause_url'] == "http://localhost:8000/api/v1/checks/" + str(self.a1.code) + "/pause"
+
+        assert self.a2.timeout == td(seconds=86400)
+        assert self.a2.grace == td(seconds=3600)
+        assert self.a2.status == "up"
+        assert self.a2.last_ping == self.now
+        r = self.a2.to_dict()
+        assert r['ping_url'] == "http://localhost:8000/ping/" + str(self.a2.code)
+        assert r['pause_url'] == "http://localhost:8000/api/v1/checks/" + str(self.a2.code) + "/pause"
```
Comment (ianoti) [inline hc/api/tests/test_list_checks.py]: use `self.assert` instead.

File: hc/api/tests/test_list_checks.py [hunk 12]
```diff
@@ -54,3 +72,12 @@ def test_it_shows_only_users_checks(self):
             self.assertNotEqual(check["name"], "Bob 1")
 
     ### Test that it accepts an api_key in the request
+    def get_checks(self):
+        return self.client.get("/api/v1/checks/")
+
+    def test_it_accepts_api_key_in_request(self):
+        r = self.get_checks()
+        data = r.json()
+        self.assertTrue(data['error'], 'wrong api_key')
```
Comment (ianoti) [inline hc/api/tests/test_list_checks.py]: should be assertEqual here

**Review by ianoti — COMMENTED** [review]
I've left some changes and comments on the PR, good work though.

**Review by ianoti — APPROVED** [review]
I'm merging this in

**Pull request merged**